### DISCUSSION TITLE
LADSPA conflict

### DISCUSSION THREAD

**Discussion by kybos** [muse-sequencer/muse#1193] [Team Posts]
Hi,

I got a crash when trying out the white noise ladspa effect (I suppose it's from the standard ladspa sdk). Then I saw that there is an ID conflict with the internal freeverb plugin:
![image](https://user-images.githubusercontent.com/10009541/104087022-f7a22680-525c-11eb-8400-d472ce178e49.png)
How is this possible? Is this a bug on our side? I can't find any sensible information on the ladspa ids in the inet.
(The crash is apparently caused by the external UI that exists for freeverb and is then loaded by the noise plugin because of the same id, and of course it doesn't fit and crashes...).

**Comment by spamatica** [comment]
Ouch, probably our plugin has the wrong id.. It's been there forever but probably not handled correctly..
I think this is the relevant document http://ladspa.org/ladspa_sdk/unique_ids.html
Can't remember if this ui does anything necessary, I would think no, so maybe that is the quick fix, remove the ui.
Though how do we instantiate ladspa plugins from songfiles? If we only go by id we can't restore a song with the "wrong" one

**Comment by kybos** [comment]
Thanks for the link - so the IDs have to be requested by e-mail. Were ours possibly not properly requested? The owner of the conflicting ID seems to be the ladspa author himself...

**Comment by spamatica** [comment]
I think Werner did this port of freeverb close to 20 years ago and it might very well be he picked the last known ID+1 but neglected to report it properly...

The good news is that it's not (at least should not be) installed into a public location so we should be able to detect this and add a special case for when 1050 is loaded into MusE if needed.

**Comment by terminator356** [comment]
I've known about that problem for years.
There is also our double chorus (1051) and pan delay (1052).

There's something else too: We have a custom UI feature (from way before DSSI came along)
 that allows users to create a custom UI for any LADSPA plugin. See share/plugins.
One of the two custom UIs is for ID 1050 - meant for our freeverb plugin.
So it will try to load for plugin 1050.
I've seen it try to load with the real plugin 1050, which is not good of course.
It can safely be removed.
The other custom UI is for Tap Reverberator (2142). We are OK with that one. It's actually nice.

**Comment by terminator356** [comment]
> I've seen it try to load with the real plugin 1050, which is not good of course.

Verified. Load the noise plugin 1050 and you are presented with three sliders that were
 meant for the freeverb plugin, which actually only has one parameter - which is level !

That implementation of freeverb is old. I think freeverb moved on to freeverb III.

The custom 1050 UI can be removed. It's not very useful with just three sliders, and looks
 about the same as if the custom UI was not there anyway.

The 2142 UI really is useful, though. Makes Tap Reverberator look nice,
 with its enumerated 'reverb type' drop down list. The plugin would be difficult to use without it.

**Comment by kybos** [comment]
I removed the custom UI for 1050, since in my case loading it for the noise plugin crashed MusE, as mentioned above.
